fix JSON encoding of CRS type (#610)

geotom opened the pull request:
# Overview
Fixed the JSON encoding of the CRS type which crashed an execution request when the request was serialized

# Related Issue / Discussion
This should fix the problem described in #610 

# Additional Information
The type `owslib.crs.Crs` cannot be encoded by the default JSON encoding

# Contribution Agreement

(as per https://github.com/geopython/pywps/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [X] I'd like to contribute [feature X|bugfix Y|docs|something else] to PyWPS. I confirm that my contributions to PyWPS will be compatible with the PyWPS license guidelines at the time of contribution.
- [ ] I have already previously agreed to the PyWPS Contributions and Licensing Guidelines

cehbrecht reviewed (changes requested):
Could you please add a test for the bbox?

geotom commented:
Yes, I will add a coverage test for this. Latest next week

geotom commented:
### New changes to my branch: [issue-610](https://github.com/geotom/pywps/tree/issue-610)

I have added tests and also fixed the issue at the source of the problem, which was:
When parsing BBox inputs, a owslib `BoundingBox `was created and its crs property was simply set as input['crs']. The problem is that the crs propery of the `BoundingBox` is of the type `owslib.crs.Crs`. This lead to a failing JSONEncoder. 
I removed the check in the JSONEncoder and made now sure that the crs is properly set as string when doing the input parsing.

Should I create a new PR for this @cehbrecht or is it enought?

cehbrecht commented:
@geotom @tomkralidis should we use `getcodeurn()` instead of `str(bbox.crs)`?:
https://github.com/geopython/OWSLib/blob/2f55bee4514ab114ab43b5495d719c66fd5dce79/owslib/crs.py#L1848

geotom commented:
The difference between `getcodeurn()` instead of `str(bbox.crs)` is probably cosmetic as `str()` indirectly calls exactly this method. Not sure what you'd prefer.

tomkralidis commented:
+1 to use `getcodeurn()`.

cehbrecht commented:
@geotom ... test is failing now. `None` value of `bbox.crs` is not handled.

geotom commented:
Should not fail anymore @cehbrecht

cehbrecht commented:
@geotom thanks :)
